File: API/Dtos/ReservationWriteDto.cs
using System;
using System.Collections.Generic;

namespace API.Dtos {

    public class ReservationWriteDto {

        // PK
        public Guid ReservationId { get; set; }
        // Fks
        public int DestinationId { get; set; }
        public int PickupPointId { get; set; }
        public int PortId { get; set; }
        // Fields
        public string Date { get; set; }
        // Navigation
        public List<PassengerWriteDto> Passengers { get; set; }

    }

}

File: FrontEnd/src/app/features/check-in/classes/dtos/reservation-write-dto.ts
import { Guid } from 'guid-typescript'
// Custom
import { PassengerWriteDto } from './passenger-write-dto'

export interface ReservationWriteDto {

    // PK
    reservationId: Guid
    // Fields
    destinationId: number,
    pickupPointId: number,
    portId: number,
    date: string,
    passengers: PassengerWriteDto[]

}
